core/service.go

```diff
@@ -321,21 +329,26 @@ func (s *Service) handleReceivedMessage(msg network.Message) (err error) {
 // announce messages (block announce messages include the header but the full
 // block is required to execute `core_execute_block`).
 func (s *Service) ProcessBlockAnnounceMessage(msg network.Message) error {
+	data := msg.(*network.BlockAnnounceMessage)
+
+	header, err := types.NewHeader(data.ParentHash, data.Number, data.StateRoot, data.ExtrinsicsRoot, data.Digest)
+	if err != nil {
+		return err
+	}
 
 	// TODO: check if we should send block request message
 
-	// TODO: update message properties and use generated id
-	blockRequest := &network.BlockRequestMessage{
-		ID:            1,
-		RequestedData: 2,
-		StartingBlock: []byte{},
-		EndBlockHash:  optional.NewHash(true, common.Hash{}),
-		Direction:     1,
-		Max:           optional.NewUint32(false, 0),
+	_, err = s.blockState.GetHeader(header.Hash())
+	if err != nil && err.Error() == "Key not found" {
+		err = s.blockState.SetHeader(header)
+		if err != nil {
+			return err
+		}
+	} else {
+		return err
 	}
 
-	// send block request message to network service
-	s.msgSend <- blockRequest
```

Comment:
This is part of `TODO: check if we should send block request message`. Would be ok to delete it but might make be nice to just comment out since we will need it soon.

Reply:
we shouldn't be leaving commented out code, would rather just delete it

Reply:
rather than deleting it and adding it back in, can we just resolve it here? see above